src/shared/clean-ipc.c

```diff
@@ -41,8 +41,20 @@
 #include "macro.h"
 #include "string-util.h"
 #include "strv.h"
+#include "user-util.h"
 
-static int clean_sysvipc_shm(uid_t delete_uid) {
+static bool match_uid_gid(uid_t subject_uid, gid_t subject_gid, uid_t delete_uid, gid_t delete_gid) {
+
+        if (uid_is_valid(delete_uid) && subject_uid == delete_uid)
+                return true;
```

Comment:
If it's not valid, then it's not equal, no? So why not just `return subject_uid == delete_uid || subject_gid == delete_gid`?